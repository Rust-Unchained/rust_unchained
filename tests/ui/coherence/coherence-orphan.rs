//@ aux-build:coherence_orphan_lib.rs
#![feature(negative_impls)]

extern crate coherence_orphan_lib as lib;

use lib::TheTrait;

struct TheType;

impl TheTrait<usize> for isize {
	fn the_fn(&self) {}
}

impl TheTrait<TheType> for isize {
	fn the_fn(&self) {}
}

impl TheTrait<isize> for TheType {
	fn the_fn(&self) {}
}

impl !Send for Vec<isize> {} //~ ERROR E0321

fn main() {}
