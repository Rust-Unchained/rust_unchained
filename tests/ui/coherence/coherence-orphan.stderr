error[E0321]: cross-crate traits with a default impl, like `Send`, can only be implemented for a struct/enum type defined in the current crate
  --> $DIR/coherence-orphan.rs:22:1
   |
LL | impl !Send for Vec<isize> {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^ can't implement cross-crate trait for type in another crate

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0321`.
